pkg/scheduler/internal/cache/cache.go

```diff
@@ -699,7 +700,7 @@ func (cache *schedulerCache) GetNodeInfo(nodeName string) (*v1.Node, error) {
 
 	n, ok := cache.nodes[nodeName]
 	if !ok {
-		return nil, fmt.Errorf("node %q not found in cache", nodeName)
+		return nil, errors.NewNotFound(v1.Resource("node"), nodeName)
```

Comment:
We don't use errors.NewNotFound to wrap errors in the scheduler package.

Reply:
I am new and hence don't have the full context.  But this is the earlier code. This  used to return API errors.

 func (c *CachedNodeInfo) GetNodeInfo(id string) (*v1.Node, error) { 
 	node, err := c.Get(id) 
  
 	if apierrors.IsNotFound(err) { 
 		return nil, err 
 	}

Reply:
@draveness Please let me know if the earlier code was a mistake and newer code doesn't want to return API errors. In that case, the issue raised would become a non-issue.

Reply:
This function returns either nil or error so no need to use a specific error type. Instead, in 

https://github.com/kubernetes/kubernetes/blob/7cb4850699a4a3c1008382c8b05f245b66ce19ad/pkg/scheduler/algorithm/predicates/predicates.go#L1386-L1394

We can replace the `if apierrors.IsNotFound(err)` with just `if err != nil`.

@draveness WDYT?

Reply:
> This function returns either nil or error so no need to use a specific error type. Instead, in
> 
> https://github.com/kubernetes/kubernetes/blob/7cb4850699a4a3c1008382c8b05f245b66ce19ad/pkg/scheduler/algorithm/predicates/predicates.go#L1386-L1394
> 
> We can replace the `if apierrors.IsNotFound(err)` with just `if err != nil`.
> 
> @draveness WDYT?

+1

Reply:
Thanks @liu-cong , @draveness I have raised another PR - https://github.com/kubernetes/kubernetes/pull/84077

Reply:
You could update this one by just force pushing this branch.

Reply:
@draveness , @liu-cong I have updated the current branch. Please review.